webapp/admin_handler.go

```diff
@@ -15,38 +15,86 @@ import (
 	"github.com/web-platform-tests/wpt.fyi/shared"
 )
 
+func checkAdmin(acl shared.GitHubAccessControl, log shared.Logger, w http.ResponseWriter) bool {
+	if acl == nil {
```

Comment:
Realized this is the `nil` check for the empty/expiry cookie case. I think it makes more sense to return a http error earlier. If you do, you can remove this check here; if not, I am always good with it.

Reply:
You mean setting `http.Error` in `NewGitHubAccessControlFromRequest`? That'd require passing `http.ResponseWriter` into it. I also think setting `http.Error` in a shared function that's meant to be reusable is a bit weird.

Reply:
I meant that case where `acl, err := shared.NewGitHubAccessControlFromRequest(a, ds, r)` return `(nil,nil)` when the cookie is not set. It is the only case that `acl` being passed in here is nil. I think your [comment](https://github.com/web-platform-tests/wpt.fyi/pull/2175/files#r495510239) answered my question here - you are reserving the `err` for unexpected stuff.